app/controllers/users_controller.rb

```diff
@@ -1,21 +1,23 @@
 class UsersController < ApplicationController
+  require 'date'
+
   before_action :authenticate_user!
-  before_action :find_user, except: [:index]
+  before_action :find_user, except: [:index, :dashboard]
 
   def index; end
 
   def show; end
 
   def edit; end
 
-  def personal_info; end
+  def camp_intro; end
 
   def show_camps
     @camps = Camp.active
   end
 
   def select_camp
-    return redirect_to personal_info_user_path, notice: "You have already selected camp" if CampUser.find_by(user_id: params[:id]).present?
+    return redirect_to camp_intro_user_path, notice: "You have already selected camp" if CampUser.find_by(user_id: params[:id]).present?
```

Comment:
Please check if we can add CampUser.find_by(user_id: params[:id]) to a before_action